src/ModelInterface.py

```diff
@@ -0,0 +1,56 @@
+
+
+
+import torch
+from torch import nn
+from torch import optim
+from torchvision import datasets, transforms
+from torch.utils.data import random_split, DataLoader
+import pytorch_lightning as pl
+import torchmetrics
+from typing import Optional
+from pytorch_lightning.loggers import CSVLogger
+from importlib import import_module
+
+
+class ModelInterface(pl.LightningModule):
+    def __init__(self, model_name):
+        super().__init__()
+        self.save_hyperparameters()
+        self.load_model(model_name)
+        self.lr = 1e-2
+
+    def load_model(self, name):
+        try:
+            Model = getattr(import_module(
+                'model.' + name), name)
+        except:
```

Comment:
This is a bad practice that you are not specifying exactly what error you are expecting. Just pointing this out. No change needed here